lib/rendition.js

```diff
@@ -42,17 +102,158 @@ class Rendition {
     }
 
     size() {
-        return this.metadata[RenditionMetadata.REPO_SIZE];
+        return fs.statSync(this.path).size;
+    }
+
+    async sha1() {
+        if (!this._sha1) {
+            this._sha1 = await fileHash(this.path, HASH_ALGORITHM);
+        }
+        return this._sha1;
     }
 
-    sha1() {
-        return this.metadata[RenditionMetadata.REPO_SHA1];
+    exists() { // see and read the file
+        return fs.existsSync(this.path);
+    }
+
+    async setMimeInformation(mime=null, encoding=null){
+        // for javascript workers mostly ------------------------------------------------------
+        // explicitly set mimetype would get priority
+        if(mime){ // to handle case when this._mime is set to default DEFAULT_CONTENT_TYPE
+            this._mime = mime;
+        }
+
+        // explicitly set encoding would get priority
+        if(encoding){ // to handle case when this.encoding would have a default value
+            this._encoding = encoding;
+        }
+
+        // shellscript mostly ------------------------------------------------------
+        if(this.mimeInfoPath && await fs.pathExists(this.mimeInfoPath)){ // use the path containing the file with mime information
+            let mimeInfoContent = await fs.readFile(this.mimeInfoPath);
+            mimeInfoContent = mimeInfoContent.toString();
+            const mimeInfo = mimeInfoContent.split(";");
```

Comment:
We should probably use a library for that, we use [content-type](https://www.npmjs.com/package/content-type) in core.

Reply:
ACK.